### PR TITLE
WIP: Install the CI workflow on this repository

### PR DESCRIPTION
**Pull request by timholy**
This pull request sets up the CI workflow on this repository.

### CODE REVIEW

**Comment by timholy** [comment]
Not equivalent to the original yet, because the original also tests ImageCore

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        with:
+          path: ~/.julia/artifacts
+          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+      - uses: julia-actions/julia-buildpkg@v1
+      - name: Install ImageCore
+        shell: julia --project=. {0}
```
Comment (fredrikekre) [inline .github/workflows/CI.yml:43]:
```suggestion
        shell: julia {0}
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        run: |
+          using Pkg
+          Pkg.add(PackageSpec(; name="ImageCore", rev="master"))
```
Comment (fredrikekre) [inline .github/workflows/CI.yml:46]:
```suggestion
          using Pkg
          Pkg.develop(PackageSpec(; path=pwd()))
          Pkg.add(PackageSpec(; name="ImageCore", rev="master"))
```

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      - uses: julia-actions/julia-runtest@v1
+      - name: Test ImageCore
+        shell: julia --project=. {0}
```
Comment (fredrikekre) [inline .github/workflows/CI.yml:49]:
```suggestion
        shell: julia {0}
```

**Review by fredrikekre — COMMENTED** [review]
You can use the global env as a temp env here I think.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (fredrikekre) [reply .github/workflows/CI.yml:43]: Perhaps something like
```suggestion
        shell: julia --project=reverse-testing {0}
```
is safer. I don't know if any of the other steps use the global env.

**Comment by timholy** [comment]
Thanks for the tips, @fredrikekre!

**Pull request merged**